src/concrete/ml/quantization/quantized_module.py

```diff
@@ -271,6 +288,24 @@ def post_processing(
         """
         return values[0] if len(values) == 1 else values
 
+    def pre_processing(self, *values: numpy.ndarray) -> Tuple[numpy.ndarray, ...]:
+        """Apply pre-processing to the input values.
+
+        For quantized modules, this method builds a NumpyModule for pre-processing
+        if it doesn't exist yet, and then applies it to the input values.
+
+        Args:
+            values (numpy.ndarray): The input values to pre-process.
+
+        Returns:
+            Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]: The pre-processed values.
+        """
+        if self._onnx_preprocessing is not None:
+            if self._preprocessing_numpy is None:
+                self._preprocessing_numpy = NumpyModule(self._onnx_preprocessing)
```

Comment:
shouldn't we just load consider a `_preprocessing_numpy` attribute only, and always build it using a onnx_processing object whenever we need (ie, directly in the dump/load methods, and/or through a public method for eg tests)

Reply:
Isn't it what we do here? If _preprocessing_numpy is None we load it it. But if the onnx_preprocessing is None we shouldn't load anything.

Reply:
what I meant is that when doing this we use 2 new attributes, but in reality only one matters (see the dumps/loads), so my guess is that we can avoid make things more complicated by only considering a single attribute instead of 2

Reply:
Ok I just realized we are in the QuantizedModule and no in the NumpyModule. You are right let me do this.

Reply:
Actually I am not sure I get it. We shouldn't rebuild it everytime we need it that's very inefficient. So we need to check if there is a onnx. If there is one we can either build the numpy preprocessing or use it if it's already built.

Reply:
I have updated let me know what you think